File: tasks/day-2/erlang/present.erl
-module(present).

-export([get_square/3, get_ribon_length/3]).

get_square(Width, Length, Height) ->
    SideA = Length * Width,
    SideB = Width * Height,
    SideC = Height * Length,
    MinSide = lists:min([SideA, SideB, SideC]),
    2 * (SideA + SideB + SideC) + MinSide.

get_ribon_length(Width, Length, Height) ->
    [MinSideA, MinSideB, _] = lists:sort([Width, Length, Height]),
    Bow = Width * Length * Height,
    Ribon = (MinSideA + MinSideB) * 2,
    Bow + Ribon.

File: tasks/day-2/golang/present/present.go
package present

import (
    "sort"
)

type Present struct {
    Length, Width, Height int
}

func (p Present) GetSquare() int {
    side_a := p.Length * p.Width
    side_b := p.Width * p.Height
    side_c := p.Height * p.Length
    
    sides := []int{side_a, side_b, side_c}
    sort.Ints(sides)
    
    return 2 * (side_a + side_b + side_c) + sides[0]
}

func (p Present) GetRibonLenght() int {
    sides := []int{p.Length, p.Width, p.Height}
    sort.Ints(sides)
 
    ribon := 2 * (sides[0] + sides[1])
    bow := p.Length * p.Width * p.Height
    
    return bow + ribon
}